src/main/java/com/homihq/db2rest/auth/AuthContext.java

```diff
@@ -0,0 +1,19 @@
+package com.homihq.db2rest.auth;
+
+import com.homihq.db2rest.auth.to.AuthInfo;
+
+public abstract class AuthContext {
+    private static final ThreadLocal<AuthInfo> currentAuthInfo = new ThreadLocal<AuthInfo>();
+
+    public static void setCurrentAuthInfo(AuthInfo authInfo) {
+        currentAuthInfo.set(authInfo);
+    }
+
+    public static AuthInfo getCurrentAuthInfo() {
+        return currentAuthInfo.get();
+    }
+
+    public static void clear() {
```

Comment:
When and how is the `clear()` called?

Reply:
Clear can be called in postHandle to clear the context after the request is processed. We don't necessarily need this if spring clear/cleans the context after the request is processed. I have seen that in one of our old code we were explicitly clearing this in postHandle

Reply:
This object is not managed by spring so it won't clear it.

In filter there is no post handle. 

So you have to clear it in filter. Let's think how once the chain returns

Were you able to get this object in controller ?

May be we set this as request attribute then clearing won't be required. Later by method argument resolver get it in controller method 

Fix the other issues n open an issue to investigate this a bit.